### ISSUE TITLE
Integrate automated testing of Accessibility ARIA requirements

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#1333]
This ticket is to integrate automated testing of Accessibility ARIA requirements within the continuous integration of GlobaLeaks, to automatically spot accessibility related issues (missing labels, etc).

Some notes from https://github.com/angular/protractor/blob/master/docs/plugins.md#accessibility-plugin maybe useful
##

**Comment by evilaliv3** [comment]
The official protractor package that could be integated easily is this: https://github.com/angular/protractor-accessibility-plugin

It allows the integration with the internal tests of [Chrome](https://github.com/GoogleChrome/accessibility-developer-tools) or the integration with [Tenon.io](https://tenon.io); i will suggest to go for the first option and eventually gived the more support of Tenon.io see if they would like to sponsor us an account.

**Comment by evilaliv3** [comment]
various interesting resources are listed here: http://shashikantjagtap.net/automated-accessibility-tools/

**Comment by evilaliv3** [comment]
I've integrated the protractor plugin! :)

thanks @sjelin, i confirm that the plugin is still working also on latest protractor!

**Comment by NSkelsey** [comment]
@evilaliv3 can you add the plugin you installed to the [development environment](https://github.com/globaleaks/GlobaLeaks/wiki/setting-up-globaleaks-development-environment) wiki page?

**Comment by evilaliv3** [comment]
It is a protractor plugin installed automatically via npm and listed in packages.json.

i think that in general we could plan a page listing all the softare used and for which reason.

**Comment by evilaliv3** [comment]
The plugin is currently not anymore part of our automated testing because of an incompatibility with the latest protractor version: https://github.com/angular/protractor-accessibility-plugin/issues/27

**Comment by AnXh3L0** [comment]
@evilaliv3 based on previous comments from 2017 above, is there any up-to-date plugin that does automated testing for the frontend, available as an angular package? Do we have any information on that?

**Comment by AnXh3L0** [comment]
@evilaliv3 Regarding this ticket, we have discussed a bit previously and I was told that Angular sort of automatically generates the ARIA components we need, for the majority of the frontend? Is there any way I can check it all so that we can figure out the best way to proceed with the automated testing for this? Thanks in advance!

**Comment by evilaliv3** [comment]
Thank you for the question @AnXh3L0 

Actually are the moment:
- angular by default injects some "aria-" attributes when we use conditions but only this to declare if an element is visible not
- ng-bootstrap as well injects some attributes (mostly labels) in relation to modals, popover and date pickers.
- The rest of the accessibility (for whistleblower interfaces only is was instead manually curated; and probably will require to be manually curated to achieve maximum usability)

Within this ticket, now included in 5.1.0 i suggest the following:
- attempt to integrate cypress-axe withing our cypress tests but just during the screenshot collection in order to validate specifically some interfaces (lets say the 20% more important)
- while refining the release attempt to fix the main defects starting from the defects that apply to most the interfaces.

Let me check some of the stashed work that i have on the integration of cypress-axe and i will be back shortly.

**Comment by evilaliv3** [comment]
Commits https://github.com/globaleaks/globaleaks-whistleblowing-software/commit/3530dbaa7c74e8718f6cd5ffb4e428a353f1551a introduces integration with axe-core via cypress-axe. The change enables automated accessibility testing of the interfaces during screenshot collection.

The patch is designed not to break test execution; instead, accessibility violations are logged. This approach allows targeting a representative set of interfaces for fixing before enforcing test failures in future iterations.
I think we could close this ticket and open one dedicated to this fix effort.
